src/lib/core/style-builder/style-builder.ts

```diff
@@ -5,10 +5,22 @@
  * Use of this source code is governed by an MIT-style license that can be
  * found in the LICENSE file at https://angular.io/license
  */
-import {Injectable} from '@angular/core';
 import {StyleDefinition} from '../style-utils/style-utils';
 
-@Injectable()
+/** A class that encapsulates CSS style generation for common directives */
 export abstract class StyleBuilder {
+
+  /** Whether to cache the generated output styles */
+  shouldCache = true;
+
+  /** Build the styles given an input string and configuration object from a host */
   abstract buildStyles(input: string, parent?: Object): StyleDefinition;
+
+  /**
+   * Run a side effect computation given the input string and the computed styles
+   * from the build task and the host configuration object
+   * NOTE: This should be a no-op unless an algorithm is provided in a subclass
+   */
+  sideEffect(_input: string, _styles: StyleDefinition, _parent?: Object) {
```

Comment:
This is still not clear. Can you provide a link or example of where a sideEffect is used?
Is this essential a `notify()` operation?

Reply:
Check out the file for layout-gap. It needs to run a side effect to apply all the correct styles to the children (since `buildStyles` is synonymous with building styles only on the host)